lib/common/xml.c

```diff
@@ -786,6 +787,64 @@ pcmk__xe_set_content(xmlNode *node, const char *format, ...)
     }
 }
 
+/*!
+ * \internal
+ * \brief Check whether the first character of a string is an XML NameStartChar
+ *
+ * See https://www.w3.org/TR/xml/#NT-NameStartChar.
+ *
+ * This is almost identical to libxml2's \c xmlIsDocNameStartChar(), but they
+ * don't expose it as part of the public API.
+ *
+ * \param[in]  utf8  UTF-8 encoded string
+ * \param[out] len   Where to store size in bytes of first character in \p utf8
```

Comment:
Allow it to be NULL if the caller doesn't care about the size

Reply:
There's only one caller, and it does care about the size. I figured that's something we could add later if needed. Not hard though